atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CqlQuery.java

```diff
@@ -51,20 +51,13 @@ public void logSlowResult(KvsProfilingLogger.LoggingFunction log, Stopwatch time
     }
 
     /**
-     * Returns an object whose {@link #toString()} implementation returns a log-safe string, and does all formatting
-     * lazily. This is intended to be used for trace logging, where the {@link #toString()} method may never be called.
+     * Returns safe string representation of the query.
      */
-    public Object getLazySafeLoggableObject() {
-        return new Object() {
-            @Override
-            public String toString() {
-                String argsString = args().stream()
-                        .filter(Arg::isSafeForLogging)
-                        .map(arg -> String.format("%s = %s", arg.getName(), arg.getValue()))
-                        .collect(Collectors.joining(", "));
-
-                return safeQueryFormat() + ": " + argsString;
-            }
-        };
+    public String getSafeLog() {
+        String argsString = args().stream()
+                .filter(Arg::isSafeForLogging)
+                .map(arg -> String.format("%s = %s", arg.getName(), arg.getValue()))
+                .collect(Collectors.joining(", "));
+        return safeQueryFormat() + ": " + argsString;
```

Comment:
Nit: if this is a relatively hot path, I'd avoid String.format at all costs, and try to avoid streams. Something along these lines should substantially reduce allocations:

```suggestion
        StringBuilder buffer = new StringBuilder()
            .append(safeQueryFormat())
            .append(": ");
        boolean first = true;
        for (Arg<?> arg : args()) {
            if (arg.isSafeForLogging()) {
                if (!first) buffer.append(", ");
                first = false;
                buffer.append(arg.getName()).append(" = ").append(arg.getValue());
            }
        }
        return buffer.toString();
```

Reply:
I don't think we use Cql in hot code, mostly for creating tables and startup things. @jeremyk-91 ?